src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9210,46 +9216,45 @@ private ExpressionSyntax ParseSubExpressionCore(Precedence precedence)
                 {
                     leftOperand = ParseIsExpression(leftOperand, opToken);
                 }
-                else
+                else if (isAssignmentOperator)
                 {
-                    if (isAssignmentOperator)
+                    ExpressionSyntax rhs = opKind == SyntaxKind.SimpleAssignmentExpression && CurrentToken.Kind == SyntaxKind.RefKeyword
+                        ? rhs = CheckFeatureAvailability(ParsePossibleRefExpression(), MessageID.IDS_FeatureRefReassignment)
```

Comment:
>rhs =  [](start = 26, length = 6)

This was not changed in the PR, but why are we assigning to `rhs` in either branch?